messaging/src/test/java/org/axonframework/serialization/json/JacksonConverterTest.java

```diff
@@ -16,87 +16,99 @@
 
 package org.axonframework.serialization.json;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.axonframework.serialization.ConversionException;
+import org.axonframework.serialization.ConverterTestSuite;
 import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.provider.*;
 
-import static org.assertj.core.api.Assertions.assertThat;
+import java.io.IOException;
+import java.util.stream.Stream;
 
-class JacksonConverterTest {
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.params.provider.Arguments.*;
+import static org.mockito.Mockito.*;
 
-    private JacksonConverter converter;
+/**
+ * Test class validating the {@link JacksonConverter}.
+ *
+ * @author Allard Buijze
+ * @author Mateusz Nowak
+ * @author Steven van Beelen
+ */
+class JacksonConverterTest extends ConverterTestSuite<JacksonConverter> {
 
-    record TestEvent(String id, String name, int value) {}
-    record AnotherEvent(String userId, String description) {}
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().findAndRegisterModules();
 
-    @BeforeEach
-    void setUp() {
-        converter = new JacksonConverter();
+    @Override
+    protected JacksonConverter buildConverter() {
+        return new JacksonConverter(OBJECT_MAPPER);
     }
 
-    @Test
-    void shouldConvertTestEventToStringAndBack() {
-        // given
-        TestEvent original = new TestEvent("ID123", "TestName", 42);
-
-        // when
-        String serialized = converter.convert(original, String.class);
-        TestEvent deserialized = converter.convert(serialized, TestEvent.class);
-
-        // then
-        assertThat(deserialized).isEqualTo(original);
+    @SuppressWarnings("unused") // Used by ConverterTestSuite
+    static Stream<Arguments> supportedConversions() {
```

Comment:
Have you tried to make the methods non-static? It seems it should work. In that case, you can define abstract methods for the parameters.

Reply:
@smcvb 
It works as non-static if you add `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` annotation to the class.

Reply:
This suggestion worked wonders. I've adjusted the test implementation accordingly.